src/dist/signatures.rs

```diff
@@ -23,13 +23,8 @@ pub(crate) fn sequoia_policy() -> StandardPolicy<'static> {
     // 2023-02-01, which caused warnings to be displayed to rustup users from that day onwards.
     //
     // To keep supporting the Rust signature key, we allow the SHA-1 algorithm in rustup without a
-    // cutoff date when verifying the signature key bindings. SHA-1 data signatures are still
-    // disallowed.
-    policy.reject_hash_property_at(
-        HashAlgorithm::SHA1,
-        HashAlgoSecurity::SecondPreImageResistance,
-        None,
-    );
+    // cutoff date when verifying the signature key bindings.
+    policy.accept_hash(HashAlgorithm::SHA1);
```

Comment:
So this permits SHA1 in any context, which will permit it to be used to attack distributions, and it really is thoroughly broken for plain collision resistance. https://sequoia-pgp.org/blog/2023/02/01/202302-happy-sha1-day/

I don't know yet whether the failed signatures are:
a) a bug in the sequoia binding-case detection
b) some other place SHA1 is turning up in our distribution signatures that *is* attacker controlled

If it is (a) the fix is to work with sequoia to get their extended period fixed, or to issue new binding signatures on the keyring we're using (using keyring-linter), or both.

If it is (b) we need to cover that case once we understand it, since it means we are flat vulnerable.

Reply:
I agree with @rbtcollins, this patch allows data signatures, which definitely require collision resistance, using SHA-1.  We definitely don't want that.

It would be good to understand where the data signatures using SHA-1 are coming from and fix that.